### PR TITLE
chore: add preblocker

### PR DESCRIPTION
**Pull request by faddat**
## Description

Adds preblocker per sdk 50. 

closes: #4666

### Commit Message / Changelog Entry

```bash
chore: add preblocker
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: testing/simapp/app.go [hunk 1]
```diff
@@ -613,6 +613,10 @@ func NewSimApp(
 	app.BasicModuleManager.RegisterLegacyAminoCodec(legacyAmino)
 	app.BasicModuleManager.RegisterInterfaces(interfaceRegistry)
 
+	// NOTE: upgrade module is required to be prioritized
+	app.ModuleManager.SetOrderPreBlockers(
+		upgradetypes.ModuleName,
+	)
```
Comment (damiannolan) [inline testing/simapp/app.go:619]: Nice one! I think we can remove `upgradetypes.ModuleName` from `SetOrderBeginBlockers` below now

File: modules/capability/go.mod [hunk 2]
```diff
@@ -5,14 +5,14 @@ go 1.21
 replace github.com/syndtr/goleveldb => github.com/syndtr/goleveldb v1.0.1-0.20210819022825-2ae1ddf74ef7
 
 require (
-	cosmossdk.io/core v0.10.0
+	cosmossdk.io/core v0.11.0
```
Comment (damiannolan) [inline modules/capability/go.mod:8]: Would be nice to isolate this in a separate PR so that we can backport it cleanly

File: testing/simapp/app.go [hunk 3]
```diff
@@ -878,6 +888,7 @@ func (app *SimApp) AutoCliOpts() autocli.AppOptions {
 
 	return autocli.AppOptions{
 		Modules:               modules,
+		ModuleOptions:         runtimeservices.ExtractAutoCLIOptions(app.ModuleManager.Modules),
```
Comment (damiannolan) [inline testing/simapp/app.go:891]: Looks good to me 👍 

ref: (https://github.com/cosmos/cosmos-sdk/blob/v0.50.0-beta.0/simapp/app.go#L636-L655)

**Review by damiannolan — COMMENTED** [review]
Thanks a lot @faddat

File: modules/capability/go.mod (hunk 2 above)
Comment (damiannolan) [reply modules/capability/go.mod:8]: Done https://github.com/cosmos/ibc-go/pull/4689

File: testing/simapp/app.go (hunk 1 above)
Comment (damiannolan) [reply testing/simapp/app.go:619]: addressed in https://github.com/cosmos/ibc-go/pull/4676/commits/b8d8893e17eab0366ba5232295e0459933db61da

**Review by damiannolan — APPROVED** [review]
LGTM after the deps PRs in https://github.com/cosmos/ibc-go/pull/4691 and https://github.com/cosmos/ibc-go/pull/4689

Thank again, @faddat!

**Review by colin-axner — APPROVED** [review]
Thanks @faddat :handshake: 

I duplicated the changes for callbacks (cc @damiannolan) and opened a separate pr to bump capability module deps cleanly in the backport
